### ISSUE TITLE
Generate a drop in element to allow a user to choose the charttype

### ISSUE THREAD

**Issue by rouilj** [UMB-CS682-Team-02/tracker#57]
Currently we are using hand crafted `a` elements to allow a user to generate a chart. Create a new element to replace these.

## Step 1
First example is to add a dropdown/select that lists all the available
chart types. Then the user is able to press a submit button to chart the data shown in the index page.

To start with hard coding the chart types in the select. Once that is working and on the main branch
consider.....

## Step 2
Ideally we would not hard code the list of chart types. It should be defined in chart.py and be accessible from
the template via an extension mechanism. So registering a method like:
```
def generate_chart_selector(type="select"):
  
# do something to return html for the selector

...

def init(instance):
    instance.registerUtil('genchart', genchart)
```
https://www.roundup-tracker.org/docs/reference.html#extensions-adding-capabilities-to-your-tracker
so it can be used like:
```
<tal:block tal:contents="structure python:utils.generate_chart_selector()"></tal:block>
```
If we had a list of chart types defined at the scope of chart.py:
```
all_charttypes= { 'piechart': PieChartAction,
                              'barchart': BarChartAction,
                              ...}
```
it would make this easier. It would also allow us to replace:
```
def init(instance):
    instance.registerAction('piechart', PieChartAction)
    instance.registerAction('barchart', BarChartAction)
    instance.registerAction('stackedchart', StackedBarChartAction)
    instance.registerAction('multibarchart', MultiBarChartAction)
```
with the easier to read:
```
def init(instance):

  for charttype, chartclass in all_charttypes.items():
    instance.registerAction(charttype', chartclass)
 ```
As an added win, adding a new chart type has to be done in one place. For the tracker admin, I can upgrade chart.py with
a new heatmap chart type and make it available without any change to issue template. For the developer I don't
need to make a change to init and generate_chart_selector to add my heatmap type. Adding:

```
'heatmap': HeatmapChartAction,
```
to `all_charttypes` makes everything work.

So once step 1 of this ticket is done try step 2.

**Comment by rouilj** [comment]
This is deferred for more important tasks.
